### PR TITLE
slemicro scripts - add support for static IP

### PR DESCRIPTION
**Pull request by hardys**
This adds support for `VM_STATIC_IP` which when specified will configure a static IP via combustion.

Optionally `VM_GATEWAY_IP` may also be specified to enable operation with a non-default libvirt network.

The logic to retrieve a VM IP has also been refactored to reduce duplication, and the option to detect IP via arp added (since the default of using the DHCP lease obviously won't work in the case where we are using a static IP.

Note that this requires you first to either disable DHCP on your existing libvirt network, or create a new one with it disabled then specify via `VM_NETWORK`

### CODE REVIEW

**Comment by hardys** [comment]
@e-minguez and @alknopfler PTAL when you get a moment - I've only tested this on Linux so far, so we may need to verify this is all still working fine for Mac too

File: slemicro/README.md [hunk 1]
```diff
@@ -183,6 +183,14 @@ worker-1   Ready    <none>                      15s   v1.25.9+k3s1
 
 **TIP:** To run the VM creation in parallel you can use `for file in vm-*; do ./create_vm.sh -f ${file} &; done; wait` (the output will be a little bit messy however)
 
+### Static IPs
+
+It is possible to deploy a VM with a static IP by setting the `VM_STATIC_IP` variable.
+
```
Comment (e-minguez) [inline slemicro/README.md:189]: I'd say to also include VM_NETMASK just in case.

File: slemicro/README.md [hunk 2]
```diff
@@ -183,6 +183,14 @@ worker-1   Ready    <none>                      15s   v1.25.9+k3s1
...
+Optionally `VM_GATEWAY_IP` may also be specified (defaults to `192.168.122.1`).
+
+Note that in this configuration you must first disable DHCP for your libvirt network, which can be achieved via `virsh net-edit` to remove the `<dhcp>` stanza, then `virsh net-destroy` followed by `virsh net-start`
```
Comment (e-minguez) [inline slemicro/README.md:192]: Why is this needed? Can you just modify the libvirt network to shrink the dhcp pool?

By default it looks like:
```
  <ip address='192.168.122.1' netmask='255.255.255.0'>
    <dhcp>
      <range start='192.168.122.10' end='192.168.122.254'/>
    </dhcp>
  </ip>
```

File: slemicro/README.md (hunk 1 above)
Comment (e-minguez) [reply slemicro/README.md:189]: And maybe DNS?

File: slemicro/common.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
```
Comment (e-minguez) [inline slemicro/common.sh:1]: Thanks for doing this :)

**Review by e-minguez — COMMENTED** [review]
I've added some comments PTAL.

Also, the combustion script has this https://github.com/suse-edge/misc/blob/main/slemicro/combustion/script#L2

```
# combustion: network
```

That does this https://github.com/openSUSE/combustion/blob/master/combustion#L79-L97 and I'm not sure how this will affect this. Maybe if the VM_STATIC_IP var is set, remove that line from the combustion/script file?

File: slemicro/README.md (hunk 1 above)
Comment (hardys) [reply slemicro/README.md:189]: Sure - I was trying to keep it simple, but we can default them to sane values - I think in most cases DNS will be the same as the gateway (since DNS is provided via libvirt dnsmasq), but there may be cases where folks want to point somewhere else 👍

File: slemicro/README.md (hunk 2 above)
Comment (hardys) [reply slemicro/README.md:192]: I tried that and shrinking the DHCP pool isn't sufficient - because the then VM gets an IP via DHCP prior to the static configuration.

I could probably find a workaround for that, but the main aim here is to enable testing of DHCP-less scenarios where only static configuration is possible.

**Comment by hardys** [comment]
> I've added some comments PTAL.
> 
> Also, the combustion script has this https://github.com/suse-edge/misc/blob/main/slemicro/combustion/script#L2
> 
> ```
> # combustion: network
> ```
> 
> That does this https://github.com/openSUSE/combustion/blob/master/combustion#L79-L97 and I'm not sure how this will affect this. Maybe if the VM_STATIC_IP var is set, remove that line from the combustion/script file?

Ah thanks - perhaps that's why the shrink-DHCP-pool approach didn't work, will re-test!

Also note that I also tried [this approach](https://github.com/openSUSE/combustion#perform-modifications-in-the-initrd-environment) which requires `# combustion: network prepare` - that approach didn't work for me, but it did highlight that our current handling of combustion won't work with `prepare` because we don't pass the `--prepare` flag from the top-level `script`

I started fixing that but then discovered the approach documented doesn't seem to work (the NM configuration isn't activated prior to attempting registration) so went with the `ip` commands as recommended by @alknopfler instead

**Comment by hardys** [comment]
> That does this https://github.com/openSUSE/combustion/blob/master/combustion#L79-L97 and I'm not sure how this will affect this. Maybe if the VM_STATIC_IP var is set, remove that line from the combustion/script file?

After removing the `# combustion: network` the static configuration fails like below so I have left it as-is for now:

```
[   21.443229] shardy-SLEMicro combustion[1369]: + '[' '!' -z 192.168.123.2 ']'
[   21.443365] shardy-SLEMicro combustion[1369]: + ip addr add 192.168.123.2/24 dev eth0
[   21.448631] shardy-SLEMicro combustion[1369]: + ip route add default via 192.168.123.1
[   21.450375] shardy-SLEMicro combustion[1369]: Error: Nexthop has invalid gateway.
```

File: slemicro/README.md (hunk 1 above)
Comment (hardys) [reply slemicro/README.md:189]: added variables for gateway, prefix and DNS (all optional)
